administrator/components/com_joomlaupdate/src/Model/UpdateModel.php

```diff
@@ -85,18 +85,18 @@ public function __construct($config = [], ?MVCFactoryInterface $factory = null)
      *
      * @since    2.5.4
      */
-    public function applyUpdateSite()
+    public function applyUpdateSite(?string $updateSource = null, ?string $updateURL = null)
     {
-        // Determine the intended update URL.
-        $params = ComponentHelper::getParams('com_joomlaupdate');
+        $params       = ComponentHelper::getParams('com_joomlaupdate');
+        $updateSource = $updateSource ?: $params->get('updatesource', 'default');
+        $updateURL    = trim($updateURL ?: $params->get('customurl', 'default'));
```

Comment:
```suggestion
        $updateURL    = trim($updateURL ?: $params->get('customurl', ''));
```
"default" is not an URL, and the old code also used empty string as default in the get call.